geonode/api/tests.py

```diff
@@ -142,26 +143,31 @@ def test_dataset_get_detail_unauth_dataset_not_public(self):
         """
         list_url = reverse("api_dispatch_list", kwargs={"api_name": "api", "resource_name": "datasets"})
 
+        resp = self.client.get(list_url)
+        self.assertEqual(len(self.deserialize(resp)["objects"]), 8)
+
         layer = Dataset.objects.first()
         layer.set_permissions(self.perm_spec)
         layer.clear_dirty_state()
-        self.assertHttpNotFound(self.api_client.get(f"{list_url + str(layer.id)}/"))
 
-        self.api_client.client.login(username=self.user, password=self.passwd)
-        resp = self.api_client.get(f"{list_url + str(layer.id)}/")
-        self.assertValidJSONResponse(resp)
+        resp = self.client.get(list_url)
```

Comment:
We are not performing anymore the login, is something wanted?

Reply:
The login is on line 157, these are some more tests as unauth user